### PR TITLE
feat: Auto-height add reducers and actions

### LINKED ISSUE #12606: Task: Create reducers for storing auto-height layout for computations
Auto-height computations work well when we have a data structure based on the current DSL of the page generated and ready.

This task is to create reducers and relevant actions to store this data structure.

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
Prep for the auto height feature. 
- [x] Add a new action which updates specific canvasWidgets paths. This action doesn't trigger an eval.
- [x] Add a new reducer to store the canvas level maps. This is necessary for auto height computations
- [x] Add a new reducer to store the auto height layout tree. This is necessary for auto height computations.
- [x] Add new action functions for the new action types
- [x] Register new reducers

Fixes #12606

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manual
- Jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/actions/controlActions.tsx [hunk 1]
```diff
@@ -74,6 +75,15 @@ export const setWidgetDynamicProperty = (
   };
 };
 
+export const updateMultipleWidgetProperties = (
```
Comment (marks0351) [inline app/client/src/actions/controlActions.tsx:78]: action suffix

**Review by rahulramesha — DISMISSED** [review]

**Review by marks0351 — DISMISSED** [review]

**Review by rahulramesha — DISMISSED** [review]

**Review by marks0351 — APPROVED** [review]

**Pull request merged**